include/globals.h

```diff
@@ -893,6 +894,20 @@ constexpr std::array<T, N> to_array(const T (&arr)[N]) {
     return result;
 }
 
+// Provide a single-parameter std::accumulate overload for ranges/containers
+// This allows: auto total = std::accumulate(container);
+namespace std {
+    template <typename Range>
+    inline auto accumulate(const Range& r)
+        -> std::remove_cv_t<std::remove_reference_t<decltype(*std::begin(r))>>
+    {
+        using T = std::remove_cv_t<std::remove_reference_t<decltype(*std::begin(r))>>;
```

Comment:
`std::remove_cv_t<std::remove_reference_t<decltype(*std::begin(r))>>` can be replaced by the more concise `std::decay_t<decltype(*std::begin(r))>`.

You may have to `#include <type_traits>` to make it compile.